src/main/java/net/openhft/chronicle/bytes/internal/NativeBytesStore.java

```diff
@@ -515,11 +519,15 @@ public long readVolatileLong(long offset) {
 
     @NotNull
     @Override
-    public NativeBytesStore<U> write(
-            long offsetInRDO, byte[] bytes, int offset, int length)
-            throws IllegalStateException {
-        requireNonNull(bytes);
-        memory.copyMemory(bytes, offset, address + translate(offsetInRDO), length);
+    public NativeBytesStore<U> write(@NonNegative final long offsetInRDO,
+                                     final byte[] byteArray,
+                                     @NonNegative final int offset,
+                                     @NonNegative final  int length) throws IllegalStateException {
+        requireNonNegative(offsetInRDO);
```

Comment:
My dream is that we (and customers) can process `@NonNegative` etc at compile time using static analysis, and we can turn off checks at runtime

Reply:
This is absolutely doable. We could clone and modify the current non-null plugin.